jtuples/src/main/java/org/jtuples/Tuples.java

```diff
@@ -0,0 +1,219 @@
+            B second, C third, D fourth, E fifth) {
+        return new Quintuple<>(first, second, third, fourth, fifth);
+    }
+    
+    
+    public static <A, B, C>
+            Triple<A, B, C> flatFirst(Pair<Pair<A, B>, C> pair) {
+        return new Triple<>(
+                pair.first().first(),
+                pair.first().second(),
+                pair.second()
+        );
+    }
+    
+    public static <A, B, C>
+            Triple<A, B, C> flatSecond(Pair<A, Pair<B, C>> pair) {
+        return new Triple<>(
+                pair.first(),
+                pair.second().first(),
+                pair.second().second()
+        );
+    }
+    
+    
+    public static <A, B, C>
+            Pair<Pair<A, B>, C> associateFirst(Pair<A, Pair<B, C>> pair) {
+        return new Pair<>(
+                new Pair<>(pair.first(), pair.second().first()),
+                pair.second().second()
+        );
+    }
+    
+    public static <A, B, C>
+            Pair<A, Pair<B, C>> associateSecond(Pair<Pair<A, B>, C> pair) {
+        return new Pair<>(
+                pair.first().first(),
+                new Pair<>(pair.first().second(), pair.second())
+        );
+    }
+    
+    
+    public static <A, B> Pair<B, B> map(Pair<? extends A, ? extends A> pair,
+            Function<A, B> f) {
+        return new Pair<>(
+                f.apply(pair.first()),
+                f.apply(pair.second())
+        );
+    }
+    
+    public static <A, B> Triple<B, B, B> map(
+            Triple<? extends A, ? extends A, ? extends A> triple,
+            Function<A, B> f) {
+        return new Triple<>(
+                f.apply(triple.first()),
+                f.apply(triple.second()),
+                f.apply(triple.third())
+        );
+    }
+    
+    public static <A, B> Quadruple<B, B, B, B> map(
+            Quadruple<? extends A, ? extends A, ? extends A, ? extends A> quadruple,
+            Function<A, B> f) {
+        return new Quadruple<>(
+                f.apply(quadruple.first()),
+                f.apply(quadruple.second()),
+                f.apply(quadruple.third()),
+                f.apply(quadruple.fourth())
+        );
+    }
+    
+    public static <A, B> Quintuple<B, B, B, B, B> map(
+            Quintuple<? extends A, ? extends A, ? extends A, ? extends A, ? extends A> quintuple,
+            Function<A, B> f) {
+        return new Quintuple<>(
+                f.apply(quintuple.first()),
+                f.apply(quintuple.second()),
+                f.apply(quintuple.third()),
+                f.apply(quintuple.fourth()),
+                f.apply(quintuple.fifth())
+        );
+    }
+    
+    
+    public static <A, B> Collection<Pair<A, B>> zip(Iterable<A> firsts,
+            Iterable<B> seconds) {
+        Collection<Pair<A, B>> collection = new ArrayList<>();
+        zip(firsts.iterator(), seconds.iterator())
+                .forEachRemaining(collection::add);
+        return collection;
+    }
```

Comment:
/**
     * Zips the given iterables, resulting in a collection containing pairs of all the previous elements,
     * preserving order, and stopping at the size of the smallest of the iterators.
     * That is, for lists {@code [a, b, c]} and {@code [d, e, f, g]} the result is the list of pairs
     * {@code [(a, d), (b, e), (c, f)]}.
     */